scipy/stats/tests/test_multicomp.py

```diff
@@ -0,0 +1,165 @@
+import copy
+
+import numpy as np
+import pytest
+from numpy.testing import assert_allclose
+
+from scipy import stats
+from scipy.stats._multicomp import _pvalue_dunnett, DunnettResult
+
+
+class TestDunnett:
+
+    @pytest.mark.parametrize(
+        'rho, n_groups, df, statistic, pvalue, alternative',
+        [
+            # From Dunnett1995
+            # Tables 1a and 1b pages 1117-1118
+            (0.5, 1, 10, 1.81, 0.05, "greater"),  # different than two-sided
+            (0.5, 3, 10, 2.34, 0.05, "greater"),
+            (0.5, 2, 30, 1.99, 0.05, "greater"),
+            (0.5, 5, 30, 2.33, 0.05, "greater"),
+            (0.5, 4, 12, 3.32, 0.01, "greater"),
+            (0.5, 7, 12, 3.56, 0.01, "greater"),
+            (0.5, 2, 60, 2.64, 0.01, "greater"),
+            (0.5, 4, 60, 2.87, 0.01, "greater"),
+            (0.5, 4, 60, [2.87, 2.21], [0.01, 0.05], "greater"),
```

Comment:
It does since `_statistic_dunnett` returns an array. Or you meant something else?

Reply:
I see. I wasn't thinking clearly. You can leave it if you prefer.

It just looked out of place to test the same critical value twice. I suppose I would either 
a) have used arrays throughout to test a few critical values per number of groups/DoF OR 
b) just test critical values accuracy here and rely on tests of `dunnett` to demonstrate that it can handle arrays.

Reply:
ok I would leave it as is. The idea to me was to isolate the array mechanism from the computation of the p-value itself.